### PR TITLE
Add default argument to extra_kwargs for Faker.generate

### PR DESCRIPTION
**Pull request by schwart**
Currently have to provide extra_kwargs={} for the function to work.
Makes it slightly cleaner if the extra_kwargs argument can be omitted.
Tested in 3.6.5 and 2.7.15

This is my first pull request to a public project so apologies if I've messed up somewhere!

### CODE REVIEW

File: factory/faker.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def __init__(self, provider, locale=None, **kwargs):
         self.provider_kwargs = kwargs
         self.locale = locale
 
-    def generate(self, extra_kwargs):
+    def generate(self, extra_kwargs={}):
```
Comment (jeffwidman) [inline factory/faker.py]: Almost never a good idea to use `{}` as a default value: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

**Comment by jeffwidman** [comment]
Once I get input from @rbarrois and we decide on a solution, then @schwart it'd be great to have you update this PR and we can merge your first commit to open source 😄

File: factory/faker.py (hunk 1 above)
Comment (schwart) [reply factory/faker.py]: ah yes, rookie mistake, thanks for spotting it.

**Comment by schwart** [comment]
Thanks a load for your time, yeah I agree that ```**_kwargs``` will probably be the easiest way for most situations. Providing @rbarrois is in agreement, I'll edit the pr asap.
Thanks for the kind welcome to OSS!

**Comment by jeffwidman** [comment]
To clarify `**kwargs` should be the method param, and then inside the method use `_kwargs`... no need for `**_kwargs`

**Comment by schwart** [comment]
Updated the PR to reflect the changes that @jeffwidman requested.
Builds are now failing as ```Faker.evaluate``` tries to pass an ```extra``` dict or ```{}``` to it.
I could change the way ```Faker.evaluate``` calls ```Faker.generate```, but I don't want to get ahead of myself!
What are your thoughts on the best way to approach this?

**Comment by federicobond** [comment]
Hi there! I rebased and merged this manually. I ended up applying only the change to make the parameter optional. I left out the API change, for now, to avoid breaking current users.

Thank you @schwart and @jeffwidman! 🙌 Feel free to open up a discussion for the move to `**kwargs` in another issue.

**Comment by schwart** [comment]
ah good idea! thanks 😀

**Pull request closed without merging**